test/test_type_promotion.py

```diff
@@ -593,6 +593,19 @@ def test_sparse_true_divide(self, device, dtype):
         casting_result = dividend.to(torch.get_default_dtype()) / 2
         self.assertEqual(casting_result, torch.true_divide(dividend_sparse, 2).to_dense())
 
+    # TODO: Update this test when integer division is an error
+    @dtypes(torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64)
+    def test_integer_div_deprecated(self, device, dtype):
+        a = torch.tensor(1, device=device, dtype=dtype)
+        b = torch.tensor(1, device=device, dtype=dtype)
+        o = torch.empty(1, device=device, dtype=dtype)
+
+        with self.maybeWarnsRegex(UserWarning, '^Integer division.+is deprecated.+'):
+            c = a / b
+            c = torch.div(a, b)
+            torch.div(a, b, out=o)
```

Comment:
nit: add `out=` tests.

Reply:
I'll add some more.

Reply:
Hm... I'm not sure how much more it makes sense to add here, actually. Trying to perform division with float inputs and an integral out type will already throw a runtime error.